### PR TITLE
[OPIK-239] [UX improvements] Add charts of experiment table

### PR DESCRIPTION
**Pull request by andriidudar**
## Details
[OPIK-239] [UX improvements] Add charts of experiment table
[OPIK-223] [UX improvements] Add grouping by dataset in experiment table

image
image
image

## Issues

Resolves #

## Documentation

[OPIK-239]: https://comet-ml.atlassian.net/browse/OPIK-239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OPIK-223]: https://comet-ml.atlassian.net/browse/OPIK-223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/opik-frontend/src/components/pages/ExperimentsPage/charts/ExperimentChartContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            />
+            {chartData.lines.map((line) => {
+              const hide = hiddenLines.includes(line);
+
+              return (
+                <Line
+                  type="natural"
+                  key={line}
+                  isAnimationActive={false}
+                  dataKey={(x) => x.scores[line] || undefined}
```
Comment (ferc) [inline apps/opik-frontend/src/components/pages/ExperimentsPage/charts/ExperimentChartContainer.tsx:141]: Why we don't display the `0` data points? Not sure what we're trying to achieve with this `|| undefined` here

File: apps/opik-frontend/src/hooks/useGroupedExperimentsList.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      (acc, dataset, index) => {
+        let experimentsData = experimentsResponse[index].data;
+        if (isUndefined(experimentsData)) {
+          experimentsData = experimentsCache.current[dataset.id] ?? {
+            content: [],
+            total: 0,
+          };
+        } else {
+          experimentsCache.current[dataset.id] = experimentsData;
+        }
```
Comment (ferc) [inline apps/opik-frontend/src/hooks/useGroupedExperimentsList.ts:209]: Nice cache layer :)

**Review by ferc — APPROVED** [review]
Some improvements for next PR:
1. I guess we could avoid this kind of empty charts (dataset with no feedback scores)
Screenshot 2024-11-11 at 7 41 31 PM

2. The feedback scores with `0` as value are not displayed in the charts

File: apps/opik-frontend/src/components/pages/ExperimentsPage/charts/ExperimentChartContainer.tsx (hunk 1 above)
Comment (andriidudar) [reply apps/opik-frontend/src/components/pages/ExperimentsPage/charts/ExperimentChartContainer.tsx:141]: It was some testing code that I forgot to delete, thanks for poining me.

**Pull request merged**